libr/core/cmd_debug.c

```diff
@@ -1217,103 +1184,191 @@ static bool print_double_linked_list_bin(RCore *core,  RHeap_MallocState *main_a
+		next_node = r_agraph_add_node (g, title, chunk);
+		r_agraph_add_edge (g, prev_node, next_node);
+		r_agraph_add_edge (g, next_node, prev_node);
+		prev_node = next_node;
+	}
+	
+	r_agraph_add_edge (g, prev_node, bin_node);
+	r_agraph_add_edge (g, bin_node, prev_node);
+	r_agraph_print (g);
+
 	free (cnk);
+	r_agraph_free (g);
 
-	return true;
+	return 1;
+}
+
+static int print_double_linked_list_bin(RCore *core,  RHeap_MallocState *main_arena, ut64 m_arena, ut64 offset, ut64 num_bin, int graph) {
+	int ret = 0;
+	RListIter *iter;
+	RDebugMap *map;
+	ut64 brk_start = UT64_MAX;
+	ut64 bin = (size_t)main_arena->bins[num_bin];
+	if (!bin) {
+		return ret;
+	}
+
+	if (!core || !core->dbg || !core->dbg->maps) {
+                return ret;
+	}
+
+	r_debug_map_sync (core->dbg);
+	r_list_foreach (core->dbg->maps, iter, map) {
+		if (strstr (map->name, "heap")) {
+		        brk_start = map->addr;
+		        break;
+		}
+	}
+
+	if (brk_start == UT64_MAX) {
+		return ret;		
+	}
+	
+	bin = (size_t)m_arena + (size_t)offset + SIZE_SZ * num_bin * 2 - SIZE_SZ * 2;
+
+	switch (num_bin) {
+	case 0:
+		PRINT_GA ("  double linked list unsorted bin {\n");
+		break;
+	case 1 ... NSMALLBINS - 1:
+		PRINT_GA ("  double linked list small bin {\n");
+		break;
+	case NSMALLBINS ... NBINS - 2:
+		PRINT_GA ("  double linked list large bin {\n");
+		break;
+	}
+
+	if (!graph || graph == 1) {
+		ret = print_double_linked_list_bin_simple (core, bin, main_arena, brk_start);
+	} else {
+		ret = print_double_linked_list_bin_graph (core, bin, main_arena, brk_start);	
+	}
+
+	PRINT_GA ("\n  }\n");
+
+	return ret;
 }
 
 static void print_heap_bin(RCore *core, RHeap_MallocState *main_arena, ut64 m_arena, const char *input) {
-	int i;
+	int i, j = 2;
 	ut64 num_bin = UT64_MAX;
 	ut64 offset = (size_t)&main_arena->last_remainder - (size_t)&main_arena->mutex + SIZE_SZ;
 
 	switch (input[0]) {
-		case '\0': // "dmhb"
-			PRINT_YA ("Bins {\n");
-			for (i = 0; i < NBINS - 1; i++) {
-				PRINTF_YA (" Bin %03d:\n", i + 1);
-				if (!print_double_linked_list_bin (core, main_arena, m_arena, offset, i)) {
-					PRINT_GA ("  Empty bin");
-					PRINT_BA ("  0x0\n");
-				}
-			}
-			PRINT_YA ("\n}\n");
-			break;
-		case ' ': // "dmhb [bin_num]
-			num_bin = r_num_math (core->num, input + 1) - 1;
-			if (num_bin > NBINS - 2) {
-				eprintf ("Error: 0 < bin <= %d\n", NBINS - 1);
-				break;
-			}
-			PRINTF_YA ("  Bin %03d:\n", num_bin + 1);
-			if (!print_double_linked_list_bin (core, main_arena, m_arena, offset, num_bin)) {
-				PRINT_GA ("Empty bin");
-				PRINT_BA (" 0x0\n");
-			}
+	case '\0': // dmhb
+		PRINT_YA ("Bins {\n");
+		for (i = 0; i < NBINS - 1; i++) {
+			PRINTF_YA (" Bin %03d:\n", i + 1);
+			if (!print_double_linked_list_bin (core, main_arena, m_arena, offset, i, 0)) {
+				PRINT_GA ("  Empty bin");
+				PRINT_BA ("  0x0\n");
+			} 
+		}
+		PRINT_YA ("\n}\n");
+		break;
+	case ' ': // dmhb [bin_num]
+		j--;
```

Comment:
If theres a fallthrough add a comment like /* fallthrough */ to avoid coverity to complain and clarify that behaviour